kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/wire/WireRecord.java

```diff
@@ -36,103 +32,28 @@
 @ThreadSafe
 public class WireRecord {
 
-    /** The contained wire fields. */
-    private final List<WireField> fields;
-
-    /** The position. */
-    @Nullable
-    private final Position position;
-
-    /** The timestamp. */
-    private final Timestamp timestamp;
+    private final Map<String, TypedValue<?>> properties;
 
     /**
-     * Instantiates a new wire record.
+     * Instantiates a new {@link WireRecord}.
      *
-     * @param timestamp
-     *            the timestamp
-     * @param fields
-     *            the wire fields
+     * @param properties
+     *            Map that represents the key-value pairs
      * @throws NullPointerException
      *             if any of the argument is null
      */
-    public WireRecord(final Timestamp timestamp, final List<WireField> fields) {
-        requireNonNull(timestamp, "Timestamp cannot be null");
-        requireNonNull(fields, "Wire fields cannot be null");
+    public WireRecord(Map<String, TypedValue<?>> properties) {
+        requireNonNull(properties, "Properties cannot be null");
 
-        this.timestamp = timestamp;
-        this.position = null;
-        this.fields = Collections.unmodifiableList(fields);
+        this.properties = properties;
```

Comment:
Make it immutable. `new HashMap<>(properties)`. Otherwise clients can still be able to mutate the state of `WireRecord` using the provided map reference.